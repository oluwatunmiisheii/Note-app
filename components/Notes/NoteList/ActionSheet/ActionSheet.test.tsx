import renderer from "react-test-renderer";
import { fireEvent, render, screen } from "@testing-library/react-native";
import { BottomAction } from "./ActionSheet";

const setIsOpen = jest.fn();
const onDelete = jest.fn();

const defaultProps: React.ComponentProps<typeof BottomAction> = {
  isOpen: true,
  setIsOpen,
  onDelete,
};

const create = (props: Partial<typeof defaultProps> = defaultProps) => {
  const tree = renderer
    .create(<BottomAction {...{ ...defaultProps, ...props }} />)
    .toJSON();
  return { props, tree };
};

const setup = (props: Partial<typeof defaultProps> = defaultProps) => {
  const utils = render(<BottomAction {...{ ...defaultProps, ...props }} />);
  return { props, ...utils };
};

describe(`<${BottomAction.name}>`, () => {
  test("renders correctly", () => {
    const { tree } = create();
    expect(tree).toMatchSnapshot();
  });

  test("should not render if isOpen is false", async () => {
    setup({ isOpen: false });
    expect(screen.queryByText("Delete")).not.toBeOnTheScreen();
    expect(screen.queryByText("Cancel")).not.toBeOnTheScreen();
  });

  test("should hide actionSheet when cancel button is called", async () => {
    setup();
    const button = screen.getByText("Cancel");
    fireEvent.press(button);
    expect(setIsOpen).toBeCalledWith(false);
  });

  test("should call onDelete when delete button is pressed", async () => {
    setup();
    const button = screen.getByText("Delete");
    fireEvent.press(button);
    expect(onDelete).toBeCalled();
  });
});
