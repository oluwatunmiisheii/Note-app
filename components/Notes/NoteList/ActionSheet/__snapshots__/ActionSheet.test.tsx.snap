// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BottomAction> renders correctly 1`] = `
Array [
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.8)",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />,
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "bottom": 0,
        "height": "auto",
        "left": 0,
        "maxHeight": 933.8,
        "position": "absolute",
        "right": 0,
        "width": 750,
        "zIndex": 2,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "grey",
          "borderRadius": 10,
          "height": 5,
          "marginTop": 10,
          "width": 50,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "paddingVertical": 30,
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <RCTScrollView
        showsVerticalScrollIndicator={false}
      >
        <View>
          <View>
            <View
              style={
                Object {
                  "borderBottomColor": "#ccc",
                  "borderBottomWidth": 1,
                }
              }
            >
              <View
                accessibilityRole="button"
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderRadius": 30,
                        "height": 60,
                        "justifyContent": "center",
                        "padding": 5,
                        "width": "auto",
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    undefined,
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#6c757d",
                        "fontSize": 20,
                        "fontWeight": "500",
                        "marginLeft": 10,
                      }
                    }
                  >
                    Delete
                  </Text>
                </View>
              </View>
            </View>
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 30,
                      "height": 60,
                      "justifyContent": "center",
                      "padding": 5,
                      "width": "auto",
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  undefined,
                  undefined,
                ]
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#6c757d",
                        "fontSize": 20,
                        "fontWeight": "500",
                        "marginLeft": 10,
                      },
                      Object {
                        "color": "#007bff",
                      },
                    ]
                  }
                >
                  Cancel
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>,
]
`;
